pkg/cloud/aws/aws.go

```diff
@@ -331,7 +333,32 @@ func (a *awsProvider) findSubnet(vpcId string) (*ec2.Subnet, error) {
 			},
 		}
 	}
-	return subnets.Subnets[0], nil
+
+	for _, subnet := range subnets.Subnets {
+		output, err := a.awsClinet.DescribeInstanceTypeOfferings(&ec2.DescribeInstanceTypeOfferingsInput{
```

Comment:
it needs doc to describe the intention of this  call

Reply:
add comments in the code